src/main/java/sparksoniq/spark/iterator/flowr/ReturnClauseSparkIterator.java

```diff
@@ -49,8 +60,53 @@ public ReturnClauseSparkIterator(RuntimeTupleIterator child, RuntimeIterator exp
         return itemRDD;
     }
 
-    private JavaRDD<Item> itemRDD;
-    
-    private RuntimeTupleIterator _child;
+    @Override
+    protected boolean hasNextLocal() {
+        return _hasNext;
+    }
+
+    @Override
+    protected Item nextLocal() {
+        if(_hasNext == true){
+            Item result = _nextLocalResult;  // save the result to be returned
+            setNextLocalResult();            // calculate and store the next result
+            return result;
+        }
+        throw new IteratorFlowException("Invalid next() call in Object Lookup", getMetadata());
+    }
+
+    @Override
+    protected void openLocal(DynamicContext context) {
+        this._currentDynamicContext = context;
```

Comment:
I think that this is already done by the superclass that calls openLocal() as part of its open() implementation.

Reply:
Yes you're absolutely right. However, this left me wondering why we're passing dynamic context as a parameter to openLocal in this case. The context is already initialized and this seems redundant. Shall we remove the parameter from the function signature in HybridIterator?

Reply:
Yes, exactly! I realized this as well. Can you open a separate issue for this? We can take care of it later.